www/js/init.js

```diff
@@ -33,17 +33,168 @@
+                // Make values Boolean if 'true'/'false'
+                paramVal = paramVal === 'true' || (paramVal === 'false' ? false : paramVal);
+                setSetting(paramKey, paramVal);
+                params[paramKey] = paramVal;
+            }
+        }
+        matches = regexpUrlParams.exec(window.location.search);
+    }
+    // If we are in the PWA version launched from an extension, send a 'success' message to the extension
+    if (params.referrerExtensionURL && ~window.location.href.indexOf(params.PWAServer)) {
+        var message = '?PWA_launch=success';
+        // DEV: To test failure of the PWA, you could pause on next line and set message to '?PWA_launch=fail'
+        // Note that, as a failsafe, the PWA_launch key is set to 'fail' (in the extension) before each PWA launch
+        // so we need to send a 'success' message each time the PWA is launched
+        var frame = document.createElement('iframe');
+        frame.id = 'kiwixComm';
+        frame.style.display = 'none';
+        document.body.appendChild(frame);
+        frame.src = params.referrerExtensionURL + '/www/index.html' + message;
+        // Now remove redundant frame. We cannot use onload, because it doesn't give time for the script to run.
+        setTimeout(function () {
+            var kiwixComm = document.getElementById('kiwixComm');
+            // The only browser which does not support .remove() is IE11, but it will never run this code
+            if (kiwixComm) kiwixComm.remove();
+        }, 3000);
+    }
+})();
+
+// Since contentInjectionMode can be overriden when returning from remote PWA to extension (for example), we have to prevent an infinite loop
+// with code that warns the user to turn off the App Cache bypass in jQuery mode. Note that to turn OFF the bypass, we have to set the VALUE to true
+params.appCache = params.contentInjectionMode === 'jquery' ? true : params.appCache;
+
+/**
+ * Set the State and UI settings associated with parameters defined above
+ */
+document.getElementById('hideActiveContentWarningCheck').checked = params.hideActiveContentWarning;
+document.getElementById('disableDragAndDropCheck').checked = params.disableDragAndDrop;
+document.getElementById('showUIAnimationsCheck').checked = params.showUIAnimations;
+document.getElementById('titleSearchRange').value = params.maxSearchResultsSize;
+document.getElementById('titleSearchRangeVal').textContent = params.maxSearchResultsSize;
+document.getElementById('appThemeSelect').value = params.appTheme;
+document.getElementById('useHomeKeyToFocusSearchBarCheck').checked = params.useHomeKeyToFocusSearchBar;
+document.getElementById('openExternalLinksInNewTabsCheck').checked = params.openExternalLinksInNewTabs;
+document.getElementById('bypassAppCacheCheck').checked = !params.appCache;
+document.getElementById('appVersion').textContent = 'Kiwix ' + params.appVersion;
+
+var keyPrefix = params.keyPrefix;
```

Comment:
This line doesn't do what you think it does. The function `getSetting()`  is called before `params.keyPrefix` is defined. Although variables like this are "hoisted" (you can define them anywhere, even after a function that uses them), unfortunately that is not the case with the keys of the `params` object. Hence, you must define `params.keyPrefix` towards the very top of the code, at least before the first time that `getSetting()` is called.

You can test if your code is working by running `npm run preview`. Turn Bypass Appcache on and disable slide transition. Then press Ctrl-R and you'll see that the slide transition is turned back on if your code isn't working properly (i.e. the stored values are not being applied on startup).

Reply:
Just to demonstrate, I paused on a line that uses `keyPrefix` when rebooting the app, and you'll see from the screenshot below that it is `undefined`:

![image](https://github.com/kiwix/kiwix-js/assets/4304337/b8426e23-7166-4f39-927b-35f97fca3055)